src/objective-c/GRPCClient/private/GRPCHost.m

```diff
@@ -107,6 +115,76 @@ - (grpc_call *)unmanagedCallWithPath:(NSString *)path completionQueue:(GRPCCompl
                                   gpr_inf_future(GPR_CLOCK_REALTIME), NULL);
 }
 
+
+- (BOOL)setSSLCredentials:(nullable NSString *)pemRootCerts
+           withPrivateKey:(nullable NSString *)pemPrivateKey
+            withCertChain:(nullable NSString *)pemCertChain
+                    error:(NSError **)errorPtr {
+  static NSData *kDefaultRootsASCII;
+  static NSError *kDefaultRootsError;
+  static dispatch_once_t loading;
+  dispatch_once(&loading, ^{
+    NSString *defaultPath = @"gRPCCertificates.bundle/roots"; // .pem
+    // Do not use NSBundle.mainBundle, as it's nil for tests of library projects.
+    NSBundle *bundle = [NSBundle bundleForClass:self.class];
+    NSString *path = [bundle pathForResource:defaultPath ofType:@"pem"];
+    NSError *error;
+    // Files in PEM format can have non-ASCII characters in their comments (e.g. for the name of the
+    // issuer). Load them as UTF8 and produce an ASCII equivalent.
+    NSString *contentInUTF8 = [NSString stringWithContentsOfFile:path
+                                                        encoding:NSUTF8StringEncoding
+                                                           error:&error];
```

Comment:
So, my thought was that having the `errorPtr` is a good API, and might be helpful in the future.

Currently `grpc_ssl_credentials_create` doesn't validate its parameters, it just copies the bytes in.  But, it seems reasonable to believe in a future where there is a function that can validate the root certs / client certs / private key being correct, and therefore being able to add an error-return in a reasonable way.